WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Customs/ShippingLabelCustomsFormListViewModel.swift

```diff
@@ -49,6 +61,101 @@ final class ShippingLabelCustomsFormListViewModel: ObservableObject {
         self.stores = stores
         self.storageManager = storageManager
         self.allCountries = countries
-        self.inputViewModels = customsForms.map { .init(customsForm: $0, countries: countries) }
+        let currencySymbol: String = {
+            guard let currencyCode = CurrencySettings.CurrencyCode(rawValue: order.currency) else {
+                return ""
+            }
+            return ServiceLocator.currencySettings.symbol(from: currencyCode)
+        }()
+        self.currencySymbol = currencySymbol
+        self.inputViewModels = customsForms.map { .init(customsForm: $0, countries: countries, currency: currencySymbol) }
+
+        configureResultsControllers()
+        updateItemDetails()
+    }
+}
+
+// MARK: - Fetching and updating item details.
+//
+private extension ShippingLabelCustomsFormListViewModel {
+    /// Update item details for current customs forms
+    /// with every change in products and product variations.
+    ///
+    func updateItemDetails() {
+        $products.combineLatest($productVariations) { [weak self] (products, variations) -> [ShippingLabelCustomsForm] in
+            self?.updateCustomsForms(products: products, productVariations: variations) ?? []
+        }
+        .handleEvents(receiveOutput: { [weak self] customsForms in
+            guard let self = self else { return }
+            self.inputViewModels = customsForms.map { .init(customsForm: $0, countries: self.allCountries, currency: self.currencySymbol) }
+        })
+        .assign(to: &$customsForms)
+    }
+
+    /// Configure result controllers for products and product variations.
+    ///
+    func configureResultsControllers() {
+        resultsControllers = ShippingLabelPackageDetailsResultsControllers(siteID: order.siteID,
+                                                                           orderItems: order.items,
+                                                                           storageManager: storageManager,
+           onProductReload: { [weak self] (products) in
+            self?.products = products
+        }, onProductVariationsReload: { [weak self] (productVariations) in
+            self?.productVariations = productVariations
+        })
+
+        products = resultsControllers?.products ?? []
+        productVariations = resultsControllers?.productVariations ?? []
+    }
+
+    /// Return copy of customs forms with updated item details based on fetched products and variations.
+    ///
+    func updateCustomsForms(products: [Product], productVariations: [ProductVariation]) -> [ShippingLabelCustomsForm] {
```

Comment:
At some point here we should be sure to filter out any order items that are virtual (virtual products or virtual product variations), since those aren't shipped. Right now, those are showing up in the Package Content list for the customs form even though they aren't shown elsewhere in the shipping label flow.

Reply:
Thank you Rachel for pointing out this - I totally missed it earlier. I have fixed it in 6439bd453cf2249f46a72c78ca368cc5a675a656.